storagedriver/azure/azure.go

```diff
@@ -174,11 +193,17 @@ func (d *Driver) WriteStream(path string, offset, size int64, reader io.ReadClos
 		}
 	}
 
-	if offset != resumableOffset {
-		return storagedriver.InvalidOffsetError{Path: path, Offset: offset}
+	if offset < resumableOffset {
```

Comment:
Option 2a means the azure driver may lack support for certain use cases or may fail in odd ways on reconnect (resolved with upload retry). The s3 and filesystem drivers already implement this. Item 1 is not optional but let's take it in a separate PR.